doc/src/tutorials/physics/duffing-example.rst

```diff
@@ -0,0 +1,183 @@
+.. _duffing-example:
+
+==================================
+Duffing Oscillator with a Pendulum
+==================================
+
+In this example we demonstrate the use of functionality provided in
+:obj:`sympy.physics.mechanics` for deriving the quations of motion (EOM) for a system
+consisting of a Duffing oscillator with a pendulum.
+
+.. _fig-duffing-oscillator-pendulum:
+.. figure:: duffing-oscillator-pendulum.svg
+
+The system will be modeled using Lagrange equations. `M` is mass of the Duffing oscillator,
+`m` is mass of the pendulum, `l` is length of the pendulum. `k_1` and `k_2` are linear and
+non-linear parts of spring stiffness, and `c_1` is a viscous damping coefficient of the Duffing oscillator.
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> import sympy as sm
+   >>> import sympy.physics.mechanics as me
+   >>> from sympy.physics.mechanics.actuator import DuffingSpring
+   >>> from sympy.physics.mechanics.lagrange import LagrangesMethod
+   >>> from sympy.physics.mechanics.pathway import LinearPathway
```

Comment:
`me.<thing>` should give access to `DuffingSpring`, `LagrangesMethod`, `LinearPathway`.